appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1807,6 +1814,40 @@ private void resetLogging(String loggingProperty) {
         }
     }
 
+    /**
+     * Method to replace Env properties with corresponding value from System properties
+     * @param is InputStream from File with properties
+     * @return an instance of ByteArrayInputStream with the preprocessed properties
+     * @throws IOException
+     */
+    private ByteArrayInputStream replaceEnvProperties(InputStream is) throws IOException {
+        //preprocessing the properties read from the custom properties file
+        Properties configuration = new Properties();
+        Properties environment = new Properties();
+        configuration.load(is);
+        //set the System.getProperties to be used for the replacement process. The desired property to be mapped
+        //should need to be available on the System properties
+        environment.putAll(System.getProperties());
```

Comment:
We're replacing ENV here - we shouldn't be substituting in system properties here.
System properties replace a different regex.

This reproducer is explicitly checking `${ENV=logLevel}`

Environment replacement = `${ENV=logLevel}`
System property replacement = `${logLevel}`
MicroProfile Config replacement = `${MPCONFIG=logLevel}`
Password Alias replacement = `${ALIAS=logLevel}`

https://docs.payara.fish/community/docs/Technical%20Documentation/Payara%20Server%20Documentation/Server%20Configuration%20And%20Management/Configuration%20Options/Variable%20Substitution/Types%20of%20Variables.html

```suggestion
```

Reply:
Whether we want to hold this PR back on us not performing the other substitutions is another question, to which my answer right now is "no, that's a separate larger issue which we can follow-up on"

Reply:
Ok based on the discussion I excluded the System.getProperties call and I just leave the System.getenv() to solve ENV variables replacement